src/NUnitFramework/framework/Api/DefaultTestAssemblyBuilder.cs

```diff
@@ -99,51 +99,42 @@ private TestSuite Build(Assembly assembly, string assemblyNameOrPath, IDictionar
 
             try
             {
-                if (options.ContainsKey(FrameworkPackageSettings.DefaultTestNamePattern))
-                    TestNameGenerator.DefaultTestNamePattern = options[FrameworkPackageSettings.DefaultTestNamePattern] as string;
-
-                if (options.ContainsKey(FrameworkPackageSettings.WorkDirectory))
-                    TestContext.DefaultWorkDirectory = options[FrameworkPackageSettings.WorkDirectory] as string;
-                else
-                    TestContext.DefaultWorkDirectory = Directory.GetCurrentDirectory();
-
-                if (options.ContainsKey(FrameworkPackageSettings.TestParametersDictionary))
-                {
-                    var testParametersDictionary = options[FrameworkPackageSettings.TestParametersDictionary] as IDictionary<string, string>;
-                    if (testParametersDictionary != null)
-                    {
-                        foreach (var parameter in testParametersDictionary)
-                            TestContext.Parameters.Add(parameter.Key, parameter.Value);
-                    }
-                }
+                options.TryGetValue(FrameworkPackageSettings.DefaultTestNamePattern, out object defaultTestNamePattern);
+                TestNameGenerator.DefaultTestNamePattern = defaultTestNamePattern as string;
+                
+                options.TryGetValue(FrameworkPackageSettings.WorkDirectory, out object workDirectory);
+                TestContext.DefaultWorkDirectory = workDirectory as string ?? Directory.GetCurrentDirectory();
+
+                options.TryGetValue(FrameworkPackageSettings.TestParametersDictionary, out object testParametersDictionary);
+                if (testParametersDictionary != null)
+                    foreach (var parameter in (IDictionary<string, string>)testParametersDictionary)
+                        TestContext.Parameters.Add(parameter.Key, parameter.Value);
```

Comment:
This doesn't look like a straight refactoring, as it introduces slight behaviour changes, not sure if intented or not. Please double check.

The value of `TestNameGenerator.DefaultTestNamePattern` is set to null in case options doesn't contain the key for FrameworkPackageSettings.DefaultTestNamePattern. Prior to this change, `TestNameGenerator.DefaultTestNamePattern` wasn't touched/modified at all in this case.

For the third value `FrameworkPackageSettings.TestParametersDictionary`, prior to this change it also handled the case where the returned value was not of type `IDictionary<string, string>`. With this change, it will throw.

Reply:
As for the first problem I'll double check with the maintainers if that's acceptable behaviour, if not then I don't reckon the refactoring is even worth it at that stage, but it's not my call to make.

I'll get a commit pushed now which should address the second issue.

Thanks for the help!

Reply:
@stevenaw would the first issue described by @siprbaum here be acceptable behaviour?

Reply:
Realised I just fixed it in the latest commit anyway...never mind then! Sorry for the unnecessary mentions.